### ISSUE TITLE
LR finder for optimizing a single input tensor?

### ISSUE THREAD

**Issue by ProGamerGov** [davidtvs/pytorch-lr-finder#80]
I have an optimization task that optimizes a single tensor by passing it through a set of transforms and then into the model. Losses are then calculated by using hooks attached to various model layers.

Is is possible to use this project for finding the optimal LR for my optimization task? The code looks like it requires a DataLoader instance.

**Comment by davidtvs** [comment]
A DataLoader is not strictly needed. It requires an instance of [`TrainDataLoaderIter`](https://github.com/davidtvs/pytorch-lr-finder/blob/acc5e7ee7711a460bf3e1cc5c5f05575ba1e1b4b/torch_lr_finder/lr_finder.py#L51) and optionally [`ValDataLoaderIter`](https://github.com/davidtvs/pytorch-lr-finder/blob/acc5e7ee7711a460bf3e1cc5c5f05575ba1e1b4b/torch_lr_finder/lr_finder.py#L70).
You can implement classes that inherit from these and modify their behavior. The important thing is that the `next` method must return a pair of `input, label` which are then used for the forward pass and loss calculation.

**Comment by davidtvs** [comment]
Closing due to inactivity
